kernelci/rootfs.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (C) 2019 Collabora Limited
+# Author: Lakshmipathi G <lakshmipathi.ganapathi@collabora.com>
+#
+# This module is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation; either version 2.1 of the License, or (at your option)
+# any later version.
+#
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+from kernelci import shell_cmd
+import os
+
+
+def _build_debos(name, config, data_path):
+    cwd = os.getcwd()
+    os.chdir(data_path)
+    for arch_type in config.arch_list:
+        cmd = 'debos \
+-t architecture:{arch} \
+-t suite:{release_name} \
+-t basename:{name}/{arch} \
+-t extra_packages:"{extra_packages}" \
+-t extra_packages_remove:"{extra_packages_remove}" \
+-t extra_files_remove:"{extra_files_remove}" \
+-t script:"{script}" \
+rootfs.yaml'.format(
+            name=name,
+            data_path=data_path,
+            arch=arch_type,
+            release_name=config.debian_release,
+            extra_packages=" ".join(config.extra_packages),
+            extra_packages_remove=" ".join(config.extra_packages_remove),
+            extra_files_remove=" ".join(config.extra_files_remove),
+            script=config.script
+            )
+        ret_code = shell_cmd(cmd, True)
+        if not ret_code:
+            return False
+
+    os.chdir(cwd)
```

Comment:
If the current directory isn't changed, then why `chdir(cwd)` in this branch and not when the debos command fails?

Reply:
If we plan to support multiple rootfs build option in future (ex: `--config=build_all`), then we need change directory to original one after successfully creating first rootfs.

Reply:
I'm not asking about why we need to change the directory, I know we need to do that because the file paths in the YAML config are all relative.

My problem with this code is that it changes back to the original directory only when there was no error.  Why behave differently when there is an error?

Reply:
I assumed that  when there is an error, the kci_rootfs shouldn't proceed further and abort. If we are planning to proceed further with `next` config (ex: `--config=first_failed,next`), will add `chdir()` for that case too.

Reply:
I think we should just use "cd" in the shell command.